go/pkg/bertyprotocol/store_message.go

```diff
@@ -40,7 +44,80 @@ func (m *messageStore) setLogger(l *zap.Logger) {
 	m.logger = l.With(zap.String("group-id", fmt.Sprintf("%.6s", base64.StdEncoding.EncodeToString(m.g.PublicKey))))
 }
 
-func (m *messageStore) openMessage(ctx context.Context, e ipfslog.Entry) (*bertytypes.GroupMessageEvent, error) {
+func (m *messageStore) queueScanCache(devicePK []byte) bool {
+	select {
+	case m.cacheChan <- devicePK:
+		return true
+	default:
+		return false
+	}
+}
+
+// Add the event into a slice of the map
+func (m *messageStore) addToCache(devicePK []byte, e ipfslog.Entry) {
+	m.logger.Debug("addToCache", zap.Any("devicePK", devicePK), zap.Any("event", e))
+	m.cacheLock.Lock()
+	m.cache[string(devicePK)] = append(m.cache[string(devicePK)], e)
+	m.cacheLock.Unlock()
+
+	if !m.queueScanCache(devicePK) {
```

Comment:
addToCache must try to open messages in the cache because sometimes, a message is being added into the cache while in another goroutine, the chainkey is registered before this message has finished being added into the cache (so the cache would be empty and the message that will be added would never be opened).